Camera refactor

Daniel-Cortez opened the pull request:
This PR refactors camera code in `Player.gd`:
* Camera logic moved into a separate class (`PlayerCamera.gd`).
* Variables renamed into `lower_case`, added strict typing.
* Removed variable `look_offset`.
  This variable was used for storing the Y offset for looking up/down. The stock `Camera2D.offset` already provides this functionality.
  This brings a minor behavioral change when the camera is locked while the player looks up: if the "arrow up" button is released while the camera is locked, and then the lock time expires (`lock_time <= 0.0`), the camera descends back to the original position gradually, instead of being warped there immediately.
* Variable `adjust` was turned into a local one, as it was only used in `update()` for storing a temporary value that wasn't reused between function calls.

For now I'm not sure what to do with variables `limitLeft`, `limitRight`, `limitTop` and `limitBottom`: I can't rename them into `lower_case` notation, as the base class (`Camera2D`) already has variables `limit_left`, `limit_right`, `limit_top` and `limit_bottom`, and they are of a different type (`int`). Any help/feedback on this would be appreciated.

Daniel-Cortez committed:
Move player camera logic into a separate class

Daniel-Cortez committed:
Turn `adjust` into a local variable, as it's not used anywhere else but `post_update()`

Candescence commented:
Might be a good idea to move a bunch of other stuff out of player.gd into their own classes, assuming it doesn't meaningfully affect behavior or performance. The input stuff, for example. idk, just a potential suggestion, I suppose.

But yeah, having the camera logic be its own thing makes a lot of sense, especially since I don't think it's standard practice for the camera to be created at runtime.

DimensionWarped reviewed (commented):
Looks mostly good, but I have one gripe... we shouldn't be accessing PlayerChar members directly (yes, I know we do all over the place, but one of the main points of this refactoring effort is to cut down as much of that as I can find), we should be using an accessor function. So anywhere you have the pattern

player.camera

it should be

player.get_camera()

instead. And naturally that means you need to add an accessor function to Player.gd to retrieve the camera.

Daniel-Cortez committed:
Introduce an accessor function for `Player.camera`

Daniel-Cortez commented:
* Rebased onto the latest `Player-Refactor`.
* Reverted removal of variable `look_offset` (I wanted to use the standard `Camera2D.offset` instead, but it turns out that variable actually allows to move the camera beyond the limits set in `limit_*` variables).
* Removed redundant checks for existence of `PlayerChar.camera`, as the camera is always created for every player (besides, there already were some objects that didn't have those checks in place before doing anything with the camera).
* Introduced an accessor function for `PlayerChar.camera`, and the camera variable itself is renamed into `_camera`.

Regarding variables `limitLeft`, `limitRight`, `limitTop` and `limitBottom` I mentioned earlier, I renamed them into `target_limit_left`, `target_limit_right`, `target_limit_top` and `target_limit_bottom`. Not sure if those names reflect their purpose well, though.
